packages/polymorphic/src/Polymorphic/Polymorph.tsx

```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+
+import { usePolymorphic } from './Polymorphic.hooks';
+import {
+  PolymorphicAs,
+  PolymorphicComponentType,
+  PolymorphicPropsWithRef,
+  PolymorphicRef,
+} from './Polymorphic.types';
+
+/**
+ * Base polymorphic component
+ */
+export const BasePolymorph = <T extends PolymorphicAs = 'div'>(
+  { as, children, ...rest }: PolymorphicPropsWithRef<T>,
+  ref: PolymorphicRef<T>,
+) => {
+  const { Component } = usePolymorphic(as);
+
+  return (
+    <Component {...rest} ref={ref}>
+      {children}
+    </Component>
+  );
+};
+
+/**
+ * Polymorphic component that supports forwarded refs
+ *
+ * With the `as` prop, Polymorphic can dynamically render
+ * as any HTML element or React Component.
+ */
+export const Polymorph: PolymorphicComponentType =
+  React.forwardRef(BasePolymorph);
+Polymorph.displayName = 'Polymorph';
```

Comment:
This living in the `Polymorphic` dir is interesting. Not quite sure if there's a clear resolution.

My gut reaction is that it warrants a separate dir, but it doesn't have any accompanying styles or types files. It would feel nice to me if this file and types imported across both Polymorphic and Polymorph lived in a top-level `types.ts` file, but if this current state makes sense to everyone I'm not gonna push on changing it.

Reply:
Maybe it would have its own `.stories.tsx` file?

Reply:
This is a super weird package with respect to the directory structure guidelines. The component `Polymorph` is really secondary to the exported types, hooks and factory function.  And they're all very inter-related so a separate dir for just the component feels strange

Reply:
Either that, or if the separation is between component vs package, we have `Polymorph/Polymorph.tsx` and have the rest of the files at the package root (since they're _technically_ not related to the component, but the package)